src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOfTests.cs

```diff
@@ -361,15 +366,325 @@ End Sub
+            };
+            await testOrdinal_vb.RunAsync();
+
+            vbInput = @"
+Public Class StringOf
+    Class TestClass
+        Public Sub Main()
+            Dim Str As String = ""This is a statement""
+            If [|Str.IndexOf(value:= " + startQuote + input + startQuote + vbCharLiteral + @", comparisonType:= System.StringComparison.OrdinalIgnoreCase) = -1|] Then
+
+            End If
+        End Sub
+    End Class
+End Class
+";
+            vbFix = @"
+Public Class StringOf
+    Class TestClass
+        Public Sub Main()
+            Dim Str As String = ""This is a statement""
+            If Not Str.Contains(value:= " + startQuote + input + startQuote + vbCharLiteral + @", comparisonType:= System.StringComparison.OrdinalIgnoreCase) Then
+
+            End If
+        End Sub
+    End Class
+End Class
+";
+            testOrdinal_vb = new VerifyVB.Test
+            {
+                TestState = { Sources = { vbInput } },
+                FixedState = { Sources = { vbFix } },
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp50,
+            };
+            await testOrdinal_vb.RunAsync();
+
+            vbInput = @"
+Public Class StringOf
+    Class TestClass
+        Public Sub Main()
+            Dim Str As String = ""This is a statement""
+            If [|Str.IndexOf(" + startQuote + input + startQuote + vbCharLiteral + @", comparisonType:= System.StringComparison.OrdinalIgnoreCase) = -1|] Then
+
+            End If
+        End Sub
+    End Class
+End Class
+";
+            vbFix = @"
+Public Class StringOf
+    Class TestClass
+        Public Sub Main()
+            Dim Str As String = ""This is a statement""
+            If Not Str.Contains(" + startQuote + input + startQuote + vbCharLiteral + @", comparisonType:= System.StringComparison.OrdinalIgnoreCase) Then
+
+            End If
+        End Sub
+    End Class
+End Class
+";
+            testOrdinal_vb = new VerifyVB.Test
+            {
+                TestState = { Sources = { vbInput } },
+                FixedState = { Sources = { vbFix } },
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp50,
+            };
+            await testOrdinal_vb.RunAsync();
+
+            vbInput = @"
+Public Class StringOf
+    Class TestClass
+        Public Sub Main()
+            Dim Str As String = ""This is a statement""
+            If [|Str.IndexOf(comparisonType:= System.StringComparison.OrdinalIgnoreCase, value:= " + startQuote + input + startQuote + vbCharLiteral + @") = -1|] Then
+
+            End If
+        End Sub
+    End Class
+End Class
+";
+            vbFix = @"
+Public Class StringOf
+    Class TestClass
+        Public Sub Main()
+            Dim Str As String = ""This is a statement""
+            If Not Str.Contains(value:= " + startQuote + input + startQuote + vbCharLiteral + @", comparisonType:= System.StringComparison.OrdinalIgnoreCase) Then
```

Comment:
Note that the arguments are reversed here. I don't understand why this happens, but in the debugger I can see that `invocationOperation.SyntaxNode` (in the fixer code) has its arguments in the `(comparisonType, value)` order, but `invocationOperation.Arguments` returns it as (`value, comparisonType)`. It seems to work fine for C# though

Reply:
@pgovind Argument order in the IOperation world is in the order of runtime evaluation, does not depend on source ordering, see https://github.com/dotnet/roslyn/blob/c61052a5dda190b19abde1bab0064e1bb4bee0ce/src/Compilers/Core/Portable/Generated/Operations.Generated.cs#L541-L548

Reply:
Sounds good - feel free to file a tracking issue for VB. We can take a look at it based on customer feedback.

Reply:
https://github.com/dotnet/roslyn-analyzers/issues/3655